scripts/evaluate_predicate_classifiers.py

```diff
@@ -0,0 +1,140 @@
+
+from predicators.src import utils
+from predicators.src.approaches import create_approach
+from predicators.src.approaches.bilevel_planning_approach import \
+    BilevelPlanningApproach
+from predicators.src.envs import BaseEnv, create_new_env
+from predicators.src.main import _generate_interaction_results, \
+    _generate_or_load_offline_dataset
+from predicators.src.settings import CFG
+from predicators.src.structs import Action, Dataset, Predicate, Task
+from predicators.src.teacher import Teacher
+
+
+def _main() -> None:
+    # Parse & validate args
+    args = utils.parse_args()
+    utils.update_config(args)
+    # Create results directory.
+    outdir = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                          "results")
+    os.makedirs(outdir, exist_ok=True)
+    # Create classes. Note that seeding happens inside the env and approach.
+    env = create_new_env(CFG.env, do_cache=True)
+    # The action space and options need to be seeded externally, because
+    # env.action_space and env.options are often created during env __init__().
+    env.action_space.seed(CFG.seed)
+    for option in env.options:
+        option.params_space.seed(CFG.seed)
+    assert env.goal_predicates.issubset(env.predicates)
+    preds, _ = utils.parse_config_excluded_predicates(env)
+    # Create the train tasks.
+    train_tasks = env.get_train_tasks()
+    # If train tasks have goals that involve excluded predicates, strip those
+    # predicate classifiers to prevent leaking information to the approaches.
+    stripped_train_tasks = [
+        utils.strip_task(task, preds) for task in train_tasks
+    ]
+    # Create the agent (approach).
+    approach = create_approach(CFG.approach, preds, env.options, env.types,
+                               env.action_space, stripped_train_tasks)
+    assert approach.is_learning_based
+    assert isinstance(approach, BilevelPlanningApproach)
+    offline_dataset = _generate_or_load_offline_dataset(env, train_tasks)
+    _run_pipeline(env, approach, stripped_train_tasks, offline_dataset)
+
+
+def _run_pipeline(env: BaseEnv,
+                  approach: BilevelPlanningApproach,
+                  train_tasks: List[Task],
+                  offline_dataset: Optional[Dataset] = None) -> None:
+    assert offline_dataset is not None, "Missing offline dataset"
+    total_num_transitions = sum(
+        len(traj.actions) for traj in offline_dataset.trajectories)
+    approach.load(online_learning_cycle=None)
+    teacher = Teacher(train_tasks)
+    for i in range(CFG.num_online_learning_cycles):
+        print(f"\n\nONLINE LEARNING CYCLE {i}\n")
+        if total_num_transitions > CFG.online_learning_max_transitions:
+            break
+        interaction_requests = approach.get_interaction_requests()
+        if not interaction_requests:
+            break  # agent doesn't want to learn anything more; terminate
+        interaction_results, _ = _generate_interaction_results(
```

Comment:
is the reason that we need to go through this process just because we want to use `total_num_transitions` to break above? if so, can we instead just break whenever `load` fails? and then, can we simplify this function a lot? something like:

```
def _run_pipeline(env: BaseEnv,
                  approach: BilevelPlanningApproach) -> None:
    approach.load(online_learning_cycle=None)
    for i in range(CFG.num_online_learning_cycles):
        print(f"\n\nONLINE LEARNING CYCLE {i}\n")
        try:
            approach.load(online_learning_cycle=i)
        _evaluate_preds(
            approach._get_current_predicates(),  # pylint: disable=protected-access
            env)
        except FileNotFoundError:
            break
```